data_pipelines/resources/io_managers.py

```diff
@@ -55,7 +55,11 @@ def __init__(self, base_path: UPath):
     def dump_to_path(
         self, context: OutputContext, obj: xr.DataArray, path: UPath
     ) -> None:
-        obj.to_zarr(path, mode="w", storage_options=path.storage_options)
+        obj.to_zarr(
+            str(path),
```

Comment:
UPath also has the method `as_uri` which seems to be meant for this purpose, but I see that internally it just returns `str(self)`. 
It should also be fine to pass the UPath object directly as it inherits from pathlib.Path. Do cast it to str here to avoid confusion with the storage_options that are passed alongside?